src/Plugin/Block/ContentLanguageBlock.php

```diff
@@ -86,26 +96,18 @@ public function build() {
       return $build;
     }
 
-    $route_name = $this->pathMatcher->isFrontPage() ? '<front>' : '<current>';
-    $links = $this->languageManager->getLanguageSwitchLinks(LanguageInterface::TYPE_CONTENT, Url::fromRoute($route_name));
-
-    if (isset($links->links)) {
-      // Only show links to the available translation languages except the
-      // current one.
-      $available_languages = array_intersect_key($links->links, $entity->getTranslationLanguages());
-      unset($available_languages[$translation->language()->getId()]);
+    $available_languages = $this->contentLanguageSwitcherProvider->getAvailableEntityLanguages($entity);
 
-      $build = [
-        '#theme' => 'links__oe_multilingual_content_language_block',
-        '#links' => $available_languages,
-        '#attributes' => [
-          'class' => [
-            "language-switcher-{$links->method_id}",
-          ],
+    $build = [
+      '#theme' => 'links__oe_multilingual_content_language_block',
+      '#links' => $available_languages,
+      '#attributes' => [
+        'class' => [
+          "language-switcher",
```

Comment:
Why do we need this class here?

Reply:
Not sure why it was there in the first place ... :(